app/loaders.ts

```diff
@@ -0,0 +1,13 @@
+import { useRouteLoaderData } from '@remix-run/react';
+import type { RootLoaderData } from './root';
+
+export type LoaderData = {
+  root: RootLoaderData;
+  // Add other route loader types here
+};
+
+export function getLoaderData<K extends keyof LoaderData>(
```

Comment:
This way we can define all of our loaders and what they should return. Otherwise, in useTheme I have to type cast and say something like 
`const cookieSettings = useRouteLoaderData('root') as CookieSettings | null
Is this unnecessary  you think?

Reply:
can we not use this https://remix.run/docs/en/main/start/tutorial#type-inference?

Reply:
Yea that works. Did it here ba8bd94cc695864354eb7850a4eabd768e6547e3

My only concern with this is that [useLoaderData](https://remix.run/docs/en/main/hooks/use-loader-data) seems like its best to only use useLoaderData when in the context of the route you want loaderData from. Otherwise the "closest route" seems too arbitrary

Reply:
can't we then use useRouteLoaderData with generic to get the proper type?

Reply:
I mean `useRouteLoaderData<...>('root')`

Reply:
ahh yea, fixed